### DISCUSSION TITLE
BringIntoView not working on newly added control.

### DISCUSSION THREAD

**Discussion by Bil97** [AvaloniaUI/Avalonia#7419] [Q&A] [answered]
I am working on a project where  a user can add data dynamically. When the content does not fit on screen, The newly added content should come to view. I can't get a way of solving it.
I have tried these three methods:
```C#
textBox.BringIntoView(new Rect(new Size(textBox.Bounds.Width, textBox.Bounds.Height)));
textBox.BringIntoView(Bounds);
textBox.BringIntoView();
```
Each time the content scrolls to the first content.

Here is a sample project to demonstrate what I mean.
[Bring.zip](https://github.com/AvaloniaUI/Avalonia/files/7921535/Bring.zip)

**Comment by maxkatz6** [comment, marked as answer]
Dispatcher priority should be specified, for example "Background":
```
Dispatcher.UIThread.Post(textBox.BringIntoView, DispatcherPriority.Background);
```
Tested with your code and it works.
You need it because at the moment when you just added control to the parent, its sizes and position has not yet calculated, so it can't possibly move it. Running code inside of Dispatcher will allow to delay this operation.
